xaitk_saliency/impls/vis_sal_classifier/rise_scoring.py

```diff
@@ -0,0 +1,71 @@
+from typing import Dict, Any
+from xaitk_saliency import ImageClassifierSaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+from sklearn.preprocessing import maxabs_scale
+
+
+class RISEScoring (ImageClassifierSaliencyMapGenerator):
+    """
+    This saliency implementation transforms black-box
+    image classification scores into saliency heatmaps. This should
+    require a sequence of per-class confidences predicted on the
+    reference image, a number of per-class confidences as predicted
+    on perturbed images, as well as the masks of the reference image
+    perturbations (as would be output from a
+    `PerturbImage` implementation.
+
+    The perturbation masks used by the following implementation are
+    expected to be of type integer. Masks containing values of type
+    float are rounded to the nearest value and binarized
+    with value 1 replacing values greater than or equal to half of
+    the maximum value in mask after rounding while 0 replaces the rest.
+    """
+
+    def __init__(
+        self,
+        p1: float = 0.0,
+    ):
+        """
+        Generate RISE-based saliency maps with optional p1 de-biasing.
+
+        :param p1: De-biasing parameter based on the masking probability.
+        """
+        self.p1 = p1
```

Comment:
Would it make sense to add some assert here that p1 is in [0,1]? Similar consideration for RISEPerturbation too.

Reply:
I'll add an exception raise and a unit-test for the check.